src/Controller/ListPageRssController.php

```diff
@@ -164,12 +191,44 @@ public function build(NodeInterface $node): CacheableResponse {
    *
    * @param \Drupal\node\NodeInterface $node
    *   The node containing the list.
+   * @param \Drupal\Core\Cache\CacheableMetadata $cache_metadata
+   *   The current cache metadata.
    *
    * @return array
    *   Array of items to be rendered.
    */
-  protected function getItemList(NodeInterface $node): array {
-    return [];
+  protected function getItemList(NodeInterface $node, CacheableMetadata $cache_metadata): array {
+    $execution_result = $this->listExecutionManager->executeList($node);
+    $results = $execution_result->getResults();
+    $result_items = [];
+    foreach ($results->getResultItems() as $item) {
+      /** @var \Drupal\Core\Entity\EntityInterface $entity */
+      $entity = $item->getOriginalObject()->getEntity();
+      $cache_metadata->addCacheableDependency($entity);
+      $creation_date = $entity->get('changed')->value;
+      $result_item = [
+        '#theme' => 'oe_list_pages_rss_item',
+        '#title' => $entity->label(),
+        '#link' => $entity->toUrl('canonical', ['absolute' => TRUE]),
+        '#guid' => $entity->toUrl('canonical', ['absolute' => TRUE]),
+        '#description' => '',
+        '#item_elements' => [
+          [
```

Comment:
Let's key this array by the actual xml node element (pub date) in this case. And let's not call it `key` but `tag` no? Also, attributes needs to be an array. Matter of fact, why arent use using the `html_tag` render element here. That has all you need and you can just print it in your template without having to loop and check and stuff.